src/lib/autoinstall/autosetup_helpers.rb

```diff
@@ -126,23 +126,13 @@ def autosetup_network
       # Prevent to be called twice in case of already configured
       return if @network_configured
 
-      if Yast::Profile.current["networking"]
-        if network_before_proposal?
-          log.info("Networking setup before the proposal")
-        else
-          log.info("Networking setup at the end of first installation stage")
-        end
-
-        log.info("Importing Network settings from configuration file")
-        Yast::WFM.CallFunction("lan_auto", ["Import", Yast::Profile.current["networking"]])
-        Yast::Profile.remove_sections("networking")
-
-        # Import also the host section in order to resolve hosts only available
-        # with the network configuration and the host entry
-        if Yast::Profile.current["host"] && network_before_proposal?
-          Yast::WFM.CallFunction("host_auto", ["Import", Yast::Profile.current["host"]])
-          Yast::Profile.remove_sections("host")
-        end
+      networking_section = Yast::Profile.current.fetch("networking", {})
+      Yast::WFM.CallFunction("lan_auto", ["Import", networking_section])
+
+      # Import also the host section in order to resolve hosts only available
+      # with the network configuration and the host entry
+      if Yast::Profile.current["host"] && network_before_proposal?
+        Yast::WFM.CallFunction("host_auto", ["Import", Yast::Profile.current["host"]])
```

Comment:
when I see all that code I still think that maybe we should introduce some client object for it. So we can write something like
```
network_client = AYClient.new("host")
network_client.import(Yast::Profile.current["host"])
```

basically caller object for AutoClients

Reply:
Yes, calling WFM.CallFunction looks weird. I will try to propose something (or plan for it) apart of this PR.